File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n = len(nums)
        totalSum = 0
        maxSum = float('-inf')
        for r in range(n):
            curr = nums[r]
            totalSum += curr
            maxSum = max(maxSum, totalSum)
            if totalSum<0:
                totalSum=0


        return maxSum

File: 53-maximum-subarray/maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let currSum = 0
    let res = -Infinity
    for(let x of nums){
        currSum+= x
        res = Math.max(res,currSum)
        if (currSum<0){
            currSum = 0
        }
    }
    return res
};